runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/control/SdkHarnessClientTest.java

```diff
@@ -103,4 +173,149 @@ public void testNewBundleNoDataDoesNotCrash() throws Exception {
         BeamFnApi.InstructionResponse.newBuilder().setProcessBundle(response).build());
     activeBundle.getBundleResponse().get();
   }
+
+  @Test
+  public void testNewBundleAndProcessElements() throws Exception {
+    InProcessServerFactory serverFactory = InProcessServerFactory.create();
+    ExecutorService executor = Executors.newCachedThreadPool();
+
+    final GrpcFnServer<GrpcLoggingService> loggingServer =
+        GrpcFnServer.allocatePortAndCreateFor(
+            GrpcLoggingService.forWriter(Slf4jLogWriter.getDefault()), serverFactory);
+    GrpcDataService grpcDataService = GrpcDataService.create(executor);
+    GrpcFnServer<GrpcDataService> dataServer =
+        GrpcFnServer.allocatePortAndCreateFor(grpcDataService, serverFactory);
+
+    SynchronousQueue<FnApiControlClient> clientPool = new SynchronousQueue<>();
+    FnApiControlClientPoolService clientPoolService =
+        FnApiControlClientPoolService.offeringClientsToPool(clientPool);
+    final GrpcFnServer<FnApiControlClientPoolService> controlServer =
+        GrpcFnServer.allocatePortAndCreateFor(clientPoolService, serverFactory);
+
+    Future<Void> harness = executor.submit(() -> {
+      FnHarness.main(PipelineOptionsFactory.create(),
+          loggingServer.getApiServiceDescriptor(),
+          controlServer.getApiServiceDescriptor(),
+          new ManagedChannelFactory() {
+            @Override
+            public ManagedChannel forDescriptor(ApiServiceDescriptor apiServiceDescriptor) {
+              return InProcessChannelBuilder.forName(apiServiceDescriptor.getUrl()).build();
+            }
+          },
+          StreamObserverFactory.direct());
+      return null;
+    });
+
+    ProcessBundleDescriptor processBundleDescriptor =
+        getProcessBundleDescriptor(dataServer.getApiServiceDescriptor());
+
+    BeamFnApi.Target inputTarget =
+        BeamFnApi.Target.newBuilder()
+            .setName("local_output")
```

Comment:
runner_output and runner_input, local is relative so when the SDK harness gets this it will seem like its local to the SDK harness and not to the Runner.

Reply:
These are the constant names in the `RemoteGrpcReadNode` and `RemoteGrpcWriteNode` classes; changed these to extract from the actual nodes in the `ProcessBundleDescriptor`.